extensions/scarb-snforge-test-collector/tests/test.rs

```diff
@@ -0,0 +1,283 @@
+
+    let json: Value = serde_json::from_str(&snforge_sierra).unwrap();
+
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["block_id_type"],
+        "Tag"
+    );
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["block_id_value"],
+        "Latest"
+    );
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["url"],
+        "http://your.rpc.url"
+    );
+
+    assert_eq!(&json[0]["test_cases"][0]["name"], "forge_test::tests::test");
+}
+
+const FORK_HASH_TEST: &str = indoc! {r#"
+    #[cfg(test)]
+    mod tests {
+        #[fork(url: "http://your.rpc.url", block_id: BlockId::Hash(123))]
+        #[test]
+        fn test() {
+            assert(true == true, 'it works!')
+        }
+    }
+    "#
+};
+
+#[test]
+fn forge_test_with_fork_hash_attribute() {
+    let t = TempDir::new().unwrap();
+    let pkg1 = t.child("forge");
+
+    ProjectBuilder::start()
+        .name("forge_test")
+        .lib_cairo(FORK_HASH_TEST)
+        .build(&pkg1);
+    Scarb::quick_snapbox()
+        .arg("snforge-test-collector")
+        .current_dir(&pkg1)
+        .assert()
+        .success();
+
+    let snforge_sierra = pkg1
+        .child("target/dev/snforge/forge_test.snforge_sierra.json")
+        .read_to_string();
+
+    let json: Value = serde_json::from_str(&snforge_sierra).unwrap();
+
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["block_id_type"],
+        "Hash"
+    );
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["block_id_value"],
+        "123"
+    );
+    assert_eq!(
+        &json[0]["test_cases"][0]["fork_config"]["Params"]["url"],
+        "http://your.rpc.url"
+    );
+
+    assert_eq!(&json[0]["test_cases"][0]["name"], "forge_test::tests::test");
+}
+
+const SHOULD_PANIC_TEST: &str = indoc! {r#"
+    #[cfg(test)]
+    mod tests {
+        #[should_panic(expected: ('panic message', 'eventual second message',))]
+        #[test]
+        fn test() {
+            assert(true == true, 'it works!')
+        }
+    }
+    "#
+};
+
+#[test]
+fn forge_test_with_should_panic_message_attribute() {
+    let t = TempDir::new().unwrap();
+    let pkg1 = t.child("forge");
+
+    ProjectBuilder::start()
+        .name("forge_test")
+        .lib_cairo(SHOULD_PANIC_TEST)
+        .build(&pkg1);
+    Scarb::quick_snapbox()
+        .arg("snforge-test-collector")
+        .current_dir(&pkg1)
+        .assert()
+        .success();
+
+    let snforge_sierra = pkg1
+        .child("target/dev/snforge/forge_test.snforge_sierra.json")
+        .read_to_string();
+
+    let json: Value = serde_json::from_str(&snforge_sierra).unwrap();
+    dbg!();
+    assert_eq!(
+        &json[0]["test_cases"][0]["expected_result"]["Panics"].to_string(),
+        "{\"Exact\":[{\"value\":{\"val\":[1935763301,544040307,1634625891,112]}},{\"value\":{\"val\":[1935763301,544040307,1668247140,1814066021,1853125985,6649445]}}]}"
```

Comment:
I wonder why two shortstrings get serialized to so many values instead of just two. But if it works I guess it's fine.